Sources/SwiftDocCUtilities/Action/Actions/Init/InitAction.swift

```diff
@@ -0,0 +1,119 @@
+    ) throws {
+        self.documentationTitle = documentationTitle
+        self.includeTutorial = includeTutorial
+        self.catalogTemplate = catalogTemplate
+        catalogOutputPath = "\(catalogOutputDirectory)/\(documentationTitle).docc"
+    }
+    
+    /// Generates a documentation catalog from a catalog template
+    ///
+    /// - Parameter logHandle: The file handle that the convert and preview actions will print debug messages to.
+    public mutating func perform(logHandle: SwiftDocC.LogHandle) throws -> ActionResult {
+        try initAction()
+    }
+                            
+    func initAction() throws -> ActionResult {
+        
+        let catalogOutputURL = URL(fileURLWithPath: catalogOutputPath)
+        let fileManager = FileManager.default
+        var logHandle: LogHandle = .standardError
+        var initProblems: [Problem] = []
+        diagnosticEngine.filterLevel = .warning
+        diagnosticEngine.add(DiagnosticConsoleWriter(formattingOptions: []))
+        
+        defer {
+            diagnosticEngine.emit(initProblems)
+        }
+        
+        if fileManager.fileExists(atPath: catalogOutputPath) {
+            initProblems.append(
+                Problem(
+                    diagnostic: Diagnostic(
+                        severity: .error,
+                        identifier: "org.swift.DocumentationCatalogAlreadyExists",
+                        summary: Error.catalogAlreadyExists.errorDescription)
+                )
+            )
+            throw Error.catalogAlreadyExists
+        }
+        
+        do {
+            try fileManager.createDirectory(at: catalogOutputURL, withIntermediateDirectories: false)
+            let catalogTemplateFactory = CatalogTemplateFactory()
+            let documentationCatalog = try catalogTemplateFactory.createDocumentationCatalog(catalogTemplate, catalogTitle: documentationTitle)
+            try catalogTemplateFactory.constructCatalog(documentationCatalog, outputPath: catalogOutputPath)
+            print(
+                """
+                A new documentation catalog has been generated at \(catalogOutputPath).
+                """,
+                to: &logHandle
+            )
+        } catch {
+            // If an error occurs remove the ouptut created
+            if fileManager.fileExists(atPath: catalogOutputPath) {
+                try FileManager.default.removeItem(atPath: catalogOutputPath)
+            }
+            initProblems.append(
+                Problem(
+                    diagnostic: Diagnostic(
+                        severity: .error,
+                        identifier: "org.swift.InitActionUnexpectedError",
+                        summary: Error.cannotCreate.errorDescription
+                    )
+                )
+            )
+        }
+        
+        return ActionResult(
+            didEncounterError: !diagnosticEngine.problems.isEmpty,
+            outputs: [URL(string: catalogOutputPath)!]
```

Comment:
Creating a url `URL(string:)` without a scheme can result in unintentional network requests if this URL is passed to `Data(contentsOf:)` or `String(contentsOf:)`.
```suggestion
            outputs: [URL(fileURLWithPath: catalogOutputPath)]
```

Reply:
Changed in the commit https://github.com/apple/swift-docc/pull/719/commits/1aa87b972af7370f98feb4eb3362b1bd0b348259